src/main/java/ch/njol/skript/registrations/EventValues.java

```diff
@@ -24,336 +24,393 @@
+		 * 
+		 * @return The converter that will collect the value from the event.
 		 */
-		@Nullable
-		@SuppressWarnings("null")
-		public Class<? extends E>[] getExcludes() {
-			if (excludes != null)
-				return Arrays.copyOf(excludes, excludes.length);
-			return new Class[0];
+		Converter<E, T> getConverter() {
+			return converter;
 		}
-		
+
 		/**
-		 * Get the error message used when encountering an exclude value.
-		 * @return The error message to use when encountering an exclude
+		 * Get the class that represents the Event.
+		 * 
+		 * @return The class of the Event associated with this event value.
 		 */
-		@Nullable
-		public String getExcludeErrorMessage() {
-			return excludeErrorMessage;
+		Class<E> getEventClass() {
+			return event;
 		}
+
+		/**
+		 * Get the class that represents the value.
+		 * 
+		 * @return The class of the value associated with this event value.
+		 */
+		Class<T> getValueClass() {
+			return c;
+		}
+
 	}
-	
-	private final static List<EventValueInfo<?, ?>> defaultEventValues = new ArrayList<>(30);
+
+	private final static List<EventValueInfo<?, ?>> defaultEventValues = new ArrayList<>();
 	private final static List<EventValueInfo<?, ?>> futureEventValues = new ArrayList<>();
 	private final static List<EventValueInfo<?, ?>> pastEventValues = new ArrayList<>();
-	
+
 	/**
-	 * The past time of an event value. Represented by "past" or "former".
+	 * The past time of an event value. Represented when using "past" or "was".
 	 */
 	public static final int TIME_PAST = -1;
-	
+
 	/**
 	 * The current time of an event value.
 	 */
 	public static final int TIME_NOW = 0;
-	
+
 	/**
-	 * The future time of an event value.
+	 * The future time of an event value. Represented when using "future" or "will be".
 	 */
 	public static final int TIME_FUTURE = 1;
-	
+
 	/**
-	 * Get Event Values list for the specified time
+	 * Get event values list for the specified time.
+	 * 
 	 * @param time The time of the event values. One of
-	 * {@link EventValues#TIME_PAST}, {@link EventValues#TIME_NOW} or {@link EventValues#TIME_FUTURE}.
-	 * @return An immutable copy of the event values list for the specified time
+	 * 		{@link EventValues#TIME_PAST}, {@link EventValues#TIME_NOW} or {@link EventValues#TIME_FUTURE}.
+	 * @return An immutable copy of the event values list for the specified time.
 	 */
 	public static List<EventValueInfo<?, ?>> getEventValuesListForTime(int time) {
 		return ImmutableList.copyOf(getEventValuesList(time));
 	}
-	
+
 	private static List<EventValueInfo<?, ?>> getEventValuesList(int time) {
-		if (time == -1)
+		if (time == TIME_PAST)
 			return pastEventValues;
-		if (time == 0)
+		if (time == TIME_NOW)
 			return defaultEventValues;
-		if (time == 1)
+		if (time == TIME_FUTURE)
 			return futureEventValues;
-		throw new IllegalArgumentException("time must be -1, 0, or 1");
+		throw new IllegalArgumentException("time must be TIME_PAST, TIME_NOW, or TIME_FUTURE");
```

Comment:
Shouldn't this print the values of the time states?
```suggestion
		throw new IllegalArgumentException("time must be " + TIME_PAST + ", " + TIME_NOW + ", or " + TIME_FUTURE);
```